protocol/encoding.go

```diff
@@ -0,0 +1,147 @@
+// Defines methods/functions to encode/decode messages between client and server.
+// Currently this module supports JSON marshal/unmarshal only.
+// Protobuf would be supported in the feature.
+
+package protocol
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+var b64en = base64.StdEncoding.EncodeToString
+
+func MarshalTemporaryBinding(tb *merkletree.TemporaryBinding) ([]byte, error) {
+	if tb == nil {
+		return nil, nil
+	}
+	return json.Marshal(tb)
+}
+
+func MarshalSTR(str *merkletree.SignedTreeRoot) ([]byte, error) {
+	policies, err := json.Marshal(str.Policies)
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(&struct {
```

Comment:
>> But I think the response messages from the server to the client don't define a clear interface yet.
> 
> I'm not so sure about it. Shouldn't moving encoding.go to the server make sense? Because encoding.go contains mostly protocol's messages marshaling (Though the original idea was that we would add more encoding/decoding functions into this file when we support (say) Protobuf/etc.)

Maybe I misunderstood something, but I think whether the response messages define a clear interface and whether we move encoding to the server package are two separate issues. I think the issue that @arlolra was bringing up originally was that `encoding.go` also contains a lot of functions to marshal core data structures, which aren't protocol messages themselves, and it shouldn't be part of the protocol how the data structures are encoded to be sent to the client over the network. In other words, I think the idea would be for `encoding.go` to only contain encoding/decoding functions for message structs, and then it would also be fine to add functions for other encodings, like you suggested.

The clearer interface I was referring to is defining response messages more in terms of client-side operations (e.g. monitoring). But I don't know how much sense this makes...